### ISSUE TITLE
iam: Failed to create OIDC provider when deploying EKS in Zhy region

### ISSUE THREAD

**Issue by zxkane** [aws/aws-cdk#8185]
Failed on creating IAM OIDC provider when deploying EKS in zhy region with service account

### Reproduction Steps

Create [eks with service account enabled](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-eks-readme.html#service-accounts) in Zhy region(cn-northwest-1).

The creation was failed by below error,

```
5/15 | 10:22:06 PM | CREATE_FAILED        | Custom::AWSCDKOpenIdConnectProvider   | MyK8SCluster/OpenIdConnectProvider/Resource/Default (MyK8SClusterOpenIdConnectProvider1069C8B5) Failed to create resource. UnknownEndpoint: Inaccessible host: `iam.cn-northwest-1.amazonaws.com.cn'. This service may not be available in the `cn-northwest-1' region.
    at Request.ENOTFOUND_ERROR (/var/runtime/node_modules/aws-sdk/lib/event_listeners.js:495:46)
    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/var/runtime/node_modules/aws-sdk/lib/request.js:683:14)
    at ClientRequest.error (/var/runtime/node_modules/aws-sdk/lib/event_listeners.js:333:22)
    at ClientRequest.<anonymous> (/var/runtime/node_modules/aws-sdk/lib/http/node.js:96:19)
    at ClientRequest.emit (events.js:310:20)
    at ClientRequest.EventEmitter.emit (domain.js:482:12)
    at TLSSocket.socketErrorListener (_http_client.js:426:9)
    at TLSSocket.emit (events.js:310:20)
	new CustomResource (/eks/node_modules/@aws-cdk/core/lib/custom-resource.ts:115:21)
	\_ new OpenIdConnectProvider (/eks/node_modules/@aws-cdk/aws-iam/lib/oidc-provider.ts:120:22)
	\_ Cluster.get openIdConnectProvider [as openIdConnectProvider] (/eks/node_modules/@aws-cdk/aws-eks/lib/cluster.ts:650:37)
```

Per [IAM doc](https://docs.amazonaws.cn/en_us/general/latest/gr/iam-service.html), both zhy and bjs are using same endpoint `iam.cn-north-1.amazonaws.com.cn` for IAM.

### Error Log

See above

### Environment

  - **CLI Version      :** 1.41.0
  - **Framework Version:**
  - **OS               :**
  - **Language         :** ts

### Other

--- 

This is :bug: Bug Report

**Comment by pahud** [comment]
@zxkane Let me try fix it though I am not 100% confident but worth to explore it.

copy @eladb

**Comment by zxkane** [comment]
It also can be fixed if upstream `aws-js-sdk` accepts the request https://github.com/aws/aws-sdk-js/issues/3279.

**Comment by zxkane** [comment]
@eladb  What do you think the PR? Or do you think it should be indirectly fixed by `aws/aws-sdk-js`?

**Comment by zxkane** [comment]
The PR aws/aws-sdk-js#3274 has been merged. This issue will be resolved when cdk picks up latest aws-sdk-js with fix.

**Comment by eladb** [comment]
Since our OIDC connect provider uses the pre-installed AWS JavaScript SDK in Lambda, we will need to wait for the SDK to be released and then Lambda to pick it up. Probably within a couple of weeks.

**Comment by zxkane** [comment]
@eladb how about we also packages a fixed version `aws-sdk-js` with the OIDC provider? It might have stable behavior without depending on using the default aws sdk of lambda runtime. If you think so, I will have a try to contribute a new pr.

**Comment by zxkane** [comment]
This issue has been resolved after aws-sdk-js has been updated to 2.721.0 in nodejs runtime.
